docs/DeveloperGuide.md

```diff
@@ -1,34 +1,172 @@
+All these command classes inherits from the abstract `Command` class.
+3. The command execution can affect the Model (e.g. adding a module in ModuleList)
+4. The result of the command execution is passed back to the Ui.
+5. In addition, the command execution can also instruct the `Ui` to perform certain actions, 
+such as displaying help to the user.
+
+#### Model component
+The `Model` component
+* Stores a `Person` object that represents the user's details. (e.g. `totalCap`, `totalModuleCreditCompleted`)
+* Stores the `ModuleList` and `SemesterList` Data
+* Does not depend on any of the other three components
+
+#### Storage component
+The `Storage` component,
+* can save `personInfo` objects in csv format and read it back
+* can save the available module list in csv format and read it back
+* can save the semester list in csv format and read it back 
+
+
+### Implementation
+
+#### `Add to Semester` feature 
+The `Add to Semester` mechanism is facilitated by `AddtoSemCommand` which extends from `Command`. 
+It allows `ModuleManager` to assign a module to a semester by adding the module to a 
+`SemModulesList`, it implements the following operations:
+
+Given below is an example how the `Add to Semester` behaves at each step.
+
+##### Step 1:
+The user launches the application for the first time. The `SemesterList` will be initialized with the none 
+`SemModulesList` in it.
+
+##### Step 2:
+When users enter a add to semester command, e.g `add id/CS2113 s/4 mc/4`, this command will be parsed in `Parser`
+and then `Parser` returns a `AddToSemCommand`. `AddToSemCommand` then calls `Command#execute(SemesterList semesterList,
+ AvailableModulesList availableModulesList) `
+`(AddToSemCommand#execute(SemesterList semesterList, AvailableModulesList availableModulesList))`
+
+##### Step 3:
+`AddToSemCommand#execute()` then calls self method `AddToSemCommand#addModule()`.`AddToSemCommand#addModule()`
+ then calls `AddToSemCommand#checkModuleExist(semesterList)` to check whether the selected 
+module is already in the selected module list (i.e:`semesterList`, which is a `PriorityQueue<SemModulesList>`). 
+If the module is not in the list, `AddToSemCommand#addModule()` will check whether there is a semester list
+(i.e:`semesterModulesList`, which is a `ArrayList<SelectedModule>`) whose name is the module's semester name. 
+If the semester list exist, the module will be added to the list. 
+If not, `AddToSemCommand#addModule()` will create a new semester list and then add this module to the new list. and the
+the new semester list will be added to `semesterList` as well.
+
+#### Step 4:
+`AddToSemCommand#execute()` calls `Ui.showAddedToSemMessage(selectedModule.announceAdded())` to show the result to the 
+user
+
+The following sequence diagram shows how the `Add to Semester` operation works:
+![Sequence Diagram of Add to Semester](https://github.com/RenzoTsai/tp/blob/Update_DG/docs/UML%20img%20folder/Sequence%20Diagram%20of%20Add%20to%20Semester.png)
```

Comment:
Some comments on the sequence diagram
![Screenshot_4](https://user-images.githubusercontent.com/50647023/77994380-809a4200-735c-11ea-8b27-0a33cd5998c0.jpg)

Reply:
On a side note, the diagram doesn't show up properly when I load the website, and stays like this when I refresh. I had to right click the icon and open in a new tab to view it. Could there be a problem with the file linking?
![Screenshot_1](https://user-images.githubusercontent.com/50647023/77994637-f0a8c800-735c-11ea-88c8-8856f7df5bfa.jpg)